### ISSUE TITLE
fetch failed no metrics found should not be an Error

### ISSUE THREAD

**Issue by fgleason** [go-graphite/go-carbon#304]
we have 12 sites with local data written to each site. Data is in the format <location>.metric.  A query of the form 'target=Denver.NumSubscribers" will get sent to all locations and 11 of them will log an error.   The severity should be changed to Warning so it is possible to stop logging the messages without completely disabling logging.

**Comment by azhiltsov** [comment]
Just for sake of clarity. 
Are you using carbonserver for queries of something else?

**Comment by fgleason** [comment]
Yes we are. We had been using a graphite stack and did not get these messages in our logs. When we moved to go-carbon we started getting the errors in the logs. Since the return value is 400 graphite, and now carbonapi, also log this as an error. To mimic the behavior of graphite this should not be logging at all but I think changing this to a warning and not returning a 400 would be acceptable. 
[go-carbon.conf.txt](https://github.com/lomik/go-carbon/files/3383177/go-carbon.conf.txt)

**Comment by azhiltsov** [comment]
I agree that 404 for absent metric shouldn't be rendered as error. @grzkv @gksinghjsr what is your take on this?

Meanwhile you can disable access logs and keep the rest of logging. Below is how to do this
[[logging]]
logger = ""
file = "/var/log/go-carbon/go-carbon.log"
level = "info"
encoding = "json"
encoding-time = "iso8601"
encoding-duration = "seconds"

[[logging]]
logger = "access"
file = none
level = "info"
encoding = "json"
encoding-time = "iso8601"
encoding-duration = "seconds"

[[logging]]
logger = "stat"
file = "/var/log/go-carbon/go-carbon.log"
level = "error"
encoding = "json"
encoding-time = "iso8601"
encoding-duration = "seconds"

Does this work for you?

**Comment by fgleason** [comment]
Yes, this does work for me and it looks like I should be able to figure out how to stop the 404 errors in the carbonapi logs using the same idea. Thanks very much.

**Comment by azhiltsov** [comment]
You might consider trying zippet+api versions after this [commit](https://github.com/bookingcom/carbonapi/commit/61ab447b7443a58a538238725d58230c7f61230d) which contains the TLD filtering logic for zipper. It limits amount of go-carbon storages patrticular query will hit by learning from stores what top level metrics existing there and keeping this knowledge in a zipper cache.

**Comment by Civil** [comment]
@fgleason or you can use go-graphite/carbonapi which had the feature (TLD filtering logic and top-level cache) mentioned by @azhiltsov since several years ago. And current version of go-graphtie/carbonapi will treat not-found as "200" and empty response.

I'm also now working on error handling improvements in separate branch: https://github.com/go-graphite/carbonapi/tree/zipper_improve_errorhandling which makes this behavior more flexible and configurable (it allows you to override the status code that will be logged and returned)
